tests/run/for_in_iter.py

```diff
@@ -162,3 +162,22 @@ def for_in_gen(N):
     """
     for i in range(N):
         yield i
+
+def test_range_args(arg):
+    """
+    >>> test_range_args([1])
+    [0]
+    >>> test_range_args([1, 2])
+    [1]
+    >>> test_range_args([1, 2,  -1])
+    []
+    >>> test_range_args([1, 2, 1])
+    [1]
+    >>> test_two_args([1, 2, 1, 1])     # doctest: +ELLIPSIS
+    Traceback (most recent call last):
+    TypeError: ...
+    """
+    results = []
+    for i in range(*arg):
```

Comment:
I don't think this is testing the write thing (or at least it's only testing part of the right thing): I think the stuff in Optimize.py only handles the "known number of arguments at compile-time" case and not the runtime star-args case.

I find it helpful just to check out the test-case on the master branch and check that it actually fails without your fix (`python runtests.py for_in_iter`)